### DISCUSSION TITLE
What is the correct way to view contact pairs at time step? (MJX pipeline)

### DISCUSSION THREAD

**Discussion by i1Cps** [google/brax#518] [General]
What is the correct way to view contact pairs at time step? (MJX pipeline)

```
data.contact.geom
```

Seems to return all possible contact pairs rather then contact pairs at the timestep of "data". Should I be checking the force of each pair and verifying that its not equal to zero?

```
data.ncon
```

also returns a number similar to the maximum number of possible contact pairs (some pairs are repeated) rather than collisions detected.

I would have intuitively tried:

```
x_geom_collision_detected = jp.isin(x_geom_id_to_check, data.contact.geom.flatten())
```

But since every possible contact pair is flattened it always returns ```True```.

**Comment by btaba** [comment]
Hi @i1Cps , I recommend checking the `contact.dist`. If it's greater than zero, the contact is not "on". This is what [`get_data_into`](https://github.com/google-deepmind/mujoco/blob/4862b9e765daf2881686c806bf37a5a7930e3001/mjx/mujoco/mjx/_src/io.py#L376) is doing, as an example.
